threeML/io/model_plot.py

```diff
@@ -618,9 +617,12 @@ def tmpflux(current_value):
 
                 first_derivatives.append(this_derivate)
 
+            # add the nuisance parameter first derivatives at the end (does this assumption always hold?)
```

Comment:
We cannot assume that first derivatives are always going to be the last parameters in the model, unfortunately.

Actually, I don't understand why you need to treat the nuisance parameters differently then the other parameters... the numerical differentiation should be able to compute zero derivatives without issues. This is also needed because there might be parameters in the model itself that have the same behavior, i.e., they give zero contribution to the variance at a particular energy. So we need to be able to handle this case.

Did you find out that that the numerical differentiation does not work for the nuisance parameters?